src/lib/flatten/tsc.ts

```diff
@@ -1,48 +1,29 @@
 import * as path from 'path';
+import { SourceDescription } from 'rollup';
 import { ScriptTarget, ModuleKind, TranspileOutput, transpileModule, CompilerOptions } from 'typescript';
-import { readFile, outputJson, outputFile as fsOutputFile } from 'fs-extra';
 import { debug } from '../util/log';
 
 /**
  * Downlevels a .js file from ES2015 to ES5. Internally, uses `tsc`.
  *
- * @param inputFile
- * @param outputFile
  */
-export async function downlevelWithTsc(inputFile: string, outputFile: string): Promise<void> {
-  debug(`tsc ${inputFile} to ${outputFile}`);
-  const inputBuffer: Buffer = await readFile(inputFile);
-  const input: string = inputBuffer.toString();
+export async function downlevelWithTsc(code: string, filePath: string): Promise<SourceDescription> {
+  debug(`tsc ${filePath}`);
   const compilerOptions: CompilerOptions = {
     target: ScriptTarget.ES5,
     module: ModuleKind.ES2015,
     allowJs: true,
     sourceMap: true,
     importHelpers: true,
     downlevelIteration: true,
-    mapRoot: path.dirname(inputFile)
+    mapRoot: path.dirname(filePath)
   };
-  const transpiled: TranspileOutput = transpileModule(trimSourceMap(input.toString()), {
-    fileName: path.basename(outputFile),
-    moduleName: path.basename(outputFile, '.js'),
+  const transpiled: TranspileOutput = transpileModule(code, {
     compilerOptions
   });
 
-  const sourceMap = JSON.parse(transpiled.sourceMapText);
-  sourceMap['file'] = path.basename(outputFile);
-  sourceMap['sources'] = [path.basename(inputFile)];
-
-  await Promise.all([
-    fsOutputFile(outputFile, transpiled.outputText),
-    outputJson(`${outputFile}.map`, sourceMap, { spaces: 2 })
-  ]);
+  return {
+    code: transpiled.outputText,
+    map: transpiled.sourceMapText
+  };
 }
-
-const REGEXP = /\/\/# sourceMappingURL=.*\.js\.map/;
-const trimSourceMap = (fileContent: string): string => {
-  if (fileContent.match(REGEXP)) {
```

Comment:
Did you check the sourcemaps generated in the dist folder?  There have been issues in the past when an incorrect source map was going to fail the consumer's build.

I am going to double-check later that we're "safe" here

Reply:
They look good to me, but to be honest. I don't know what was the problem.
So, if you could check that would be great

Link to the artifacts;
https://1301-91553810-gh.circle-artifacts.com/0/integration/samples/core/dist/esm5/sample-core.js.map

Reply:
any update please?

Reply:
I am looking at some source map files with https://sokra.github.io/source-map-visualization/#custom

the ones from the jsx sample look good!

Reply:
material also. let' merge!